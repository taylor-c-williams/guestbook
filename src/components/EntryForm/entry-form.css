.entry-form {
  padding: 25px;
  margin: 35px;
  color: rgb(70, 70, 70);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

fieldset {
  margin: 25px;
}

form {
  padding: 25px;
}

input {
  margin: 10px;
}

button {
  margin: 10px;
}
