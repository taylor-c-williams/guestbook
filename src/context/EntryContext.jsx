import { createContext, useState } from 'react';

const EntryContext = createContext();
const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);
  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};
export { EntryProvider, EntryContext };
