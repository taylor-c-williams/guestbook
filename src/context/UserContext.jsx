import { createContext, useState } from 'react';

const UserContext = createContext();
const UserProvider = ({ children }) => {
  const [user, setUser] = useState();

  const login = (username, password) => {
    const loginSuccessful =
      username === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ username });
    return loginSuccessful;
  };

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};
export { UserProvider, UserContext };
