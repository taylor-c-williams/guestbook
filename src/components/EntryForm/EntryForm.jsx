import { useState } from 'react';
import { useUser } from '../../hooks/useUser';
import { useEntries } from '../../hooks/useEntries';
import './entry-form.css';

export default function EntryForm() {
  const [guestEntry, setGuestEntry] = useState('');
  // const { entries, setEntries } = useEntries();
  const { user, setUser } = useUser();

  // const newUserInput = (
  //   <input
  //     id="name"
  //     value={name}
  //     type="text"
  //     placeholder="Enter your Name"
  //     onChange={(e) => setName(e.target.value)}
  //   />
  // );

  // const handleSubmit = (e) => {
  //   e.preventDefault();
  //   if (!guestEntry) return;
  //   // setUser(name);
  //   setEntries([...entries, { user, message: guestEntry }]);
  //   setGuestEntry('');
  // };

  return (
    <section className="entry-form">
      <fieldset>
        <form>
          Hello, {user}
          <textarea
            id="guestEntry"
            value={guestEntry}
            placeholder="Write your message here"
            onChange={(e) => setGuestEntry(e.target.value)}
          />
          <button type="submit">Submit!</button>
        </form>
      </fieldset>
    </section>
  );
}
