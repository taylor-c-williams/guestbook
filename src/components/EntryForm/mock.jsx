import { useState } from 'react';
import { useEntries } from '../../context/EntryContext';
import { useUser } from '../../context/UserContext';
import './GuestBook.css';

export default function GuestBook() {
  const [name, setName] = useState('');
  const [guestEntry, setGuestEntry] = useState('');
  const { entries, setEntries } = useEntries();
  const { user, setUser } = useUser();

  function updateGuestName() {
    if (!guestEntry) return;
    // set username here
    setUser(name);
    setEntries([...entries, { name, message: guestEntry }]);
    setGuestEntry('');
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    updateGuestName();
  };

  const guestNameInput = (
    <input
      id="guestName"
      type="text"
      placeholder="Guest Name..."
      value={name}
      onChange={(e) => setName(e.target.value)}
    />
  );

  const displayMessage = user
    ? `Thanks for Signing ${user}!`
    : 'Please Sign the Guestbook!';

  return (
    <>
      {displayMessage}
      <form onSubmit={handleSubmit}>
        {user ? null : guestNameInput}

        <textarea
          id="guestEntry"
          value={guestEntry}
          placeholder="Your Entry!"
          onChange={(e) => setGuestEntry(e.target.value)}
        />

        <button className="button" type="submit">
          Sign 🖊️
        </button>
        {user && (
          <button
            type="button"
            onClick={() => {
              setUser('');
              setName('');
            }}
          >
            Not {user} ❓
          </button>
        )}
      </form>
    </>
  );
}
