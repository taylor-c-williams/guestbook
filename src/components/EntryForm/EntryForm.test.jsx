import React from 'react';
import { render } from '@testing-library/react';
import EntryForm from './EntryForm';
import { UserContext } from '../../context/UserContext';

it('should render the Home view', () => {
  const { container } = (
    <UserContext>
      <EntryForm />
    </UserContext>
  );
  expect(container).toMatchSnapshot();
});
