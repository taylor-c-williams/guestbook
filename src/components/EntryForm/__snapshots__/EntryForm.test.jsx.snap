// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Auth view 1`] = `undefined`;
